x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/embeddings/AzureOpenAiEmbeddingsServiceSettings.java

```diff
@@ -0,0 +1,339 @@
+        String apiVersion,
+        @Nullable Integer dimensions,
+        @Nullable String encodingFormat,
+        @Nullable Integer maxInputTokens
+    ) {}
+
+    private final String resourceName;
+    private final String deploymentId;
+    private final String apiVersion;
+    private final Integer dimensions;
+    private final Boolean dimensionsSetByUser;
+    private final String encodingFormat;
+    private final Boolean encodingFormatSetByUser;
+    private final Integer maxInputTokens;
+
+    public AzureOpenAiEmbeddingsServiceSettings(
+        String resourceName,
+        String deploymentId,
+        String apiVersion,
+        @Nullable Integer dimensions,
+        Boolean dimensionsSetByUser,
+        @Nullable String encodingFormat,
+        Boolean encodingFormatSetByUser,
+        @Nullable Integer maxInputTokens
+    ) {
+        this.resourceName = resourceName;
+        this.deploymentId = deploymentId;
+        this.apiVersion = apiVersion;
+        this.dimensions = dimensions;
+        this.dimensionsSetByUser = Objects.requireNonNull(dimensionsSetByUser);
+        this.encodingFormat = encodingFormat;
+        this.encodingFormatSetByUser = Objects.requireNonNull(encodingFormatSetByUser);
+        this.maxInputTokens = maxInputTokens;
+    }
+
+    public AzureOpenAiEmbeddingsServiceSettings(StreamInput in) throws IOException {
+        resourceName = in.readString();
+        deploymentId = in.readString();
+        apiVersion = in.readString();
+        dimensions = in.readOptionalVInt();
+        dimensionsSetByUser = in.readBoolean();
+        encodingFormat = in.readOptionalString();
+        encodingFormatSetByUser = in.readBoolean();
+        maxInputTokens = in.readOptionalVInt();
+    }
+
+    private AzureOpenAiEmbeddingsServiceSettings(CommonFields fields, Boolean dimensionsSetByUser, Boolean encodingFormatSetByUser) {
+        this(
+            fields.resourceName,
+            fields.deploymentId,
+            fields.apiVersion,
+            fields.dimensions,
+            dimensionsSetByUser,
+            fields.encodingFormat,
+            encodingFormatSetByUser,
+            fields.maxInputTokens
+        );
+    }
+
+    public String resourceName() {
+        return resourceName;
+    }
+
+    public String deploymentId() {
+        return deploymentId;
+    }
+
+    public String apiVersion() {
+        return apiVersion;
+    }
+
+    @Override
+    public Integer dimensions() {
+        return dimensions;
+    }
+
+    public Boolean dimensionsSetByUser() {
+        return dimensionsSetByUser;
+    }
+
+    public String encodingFormat() {
+        return encodingFormat;
+    }
+
+    public Boolean encodingFormatSetByUser() {
+        return encodingFormatSetByUser;
+    }
+
+    public Integer maxInputTokens() {
+        return maxInputTokens;
+    }
+
+    @Override
+    public DenseVectorFieldMapper.ElementType elementType() {
+        return DenseVectorFieldMapper.ElementType.FLOAT;
```

Comment:
Just a reminder, that we do choose to parse the encoding format from a request and the persisted state, then we need to use the value we parsed/defaulted here instead of forcing it to be a float here.